lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1206,7 +1206,8 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
     def _walk(b_path, b_top_level_dir):
         for b_item in os.listdir(b_path):
             b_abs_path = os.path.join(b_path, b_item)
-            b_rel_base_dir = b'' if b_path == b_top_level_dir else b_path[len(b_top_level_dir) + 1:]
+            common_prefix = os.path.commonprefix([b_top_level_dir, b_path])
+            b_rel_base_dir = b_path[len(common_prefix):].lstrip(b'/')
```

Comment:
[thought][conventional comments]: I wonder if using [`Path.relative_to()`] could make this more straightforward and eliminate such low-level mistakes...

```python
common_toplevel_path = Path(to_text(os.path.commonprefix([b_top_level_dir, b_path]), errors='surrogate_or_strict'))
relative_topdir_path = Path(b_path).relative_to(common_toplevel_path)
...
for b_item in os.listdir(b_path):
    ...
    rel_path = relative_topdir_path / to_text(b_item, errors='surrogate_or_strict')
    ...
```

[conventional comments]: https://conventionalcomments.org/
[`Path.relative_to()`]: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to

Reply:
Yup let me push something for this.

Reply:
Marking this thread as resolved.